src/Core/Snippets/Snippets.cs

```diff
@@ -149,6 +149,14 @@ public Snippet Compile(string code)
 
             try
             {
+                // strip any @EntryPoint() attributes from the snippet
+                var entryPointPattern = @"@\s*EntryPoint\s*\(\s*\)";
```

Comment:
The compiler might already have code to identify EntryPoint() in a code file, we should use the same regex (or other method) here

Reply:
Yes, ideally we could just tell the compiler to ignore these attributes, since only the compiler understands the semantics of the language and can truly do this correctly. I believe, however, that's not something the compiler exposes at the moment. @bettinaheim, can you confirm?

Reply:
I mean, you could also just discard the error that the compiler generates? The syntax tree (minus the entry point attribute) will be built regardless. 

Well, given that the EntryPoint attribute is in Core and we auto-open Core, you can actually be sure that something called @EntryPoint() is indeed the entry point attribute (and not another attribute with the same unqualified name). 
However, the compiler does some preprocessing like tracking all strings and comments and tagging all scopes before validating the tokens, and that is rather painful to reproduce as a regex. Hence my suggestion above.

Reply:
Yes, we could just compile and discard the error generated when multiple entry points exist (which is error QS6236). We don't currently discard/ignore any compiler errors in IQ#, but it looks like it shouldn't be that hard.

However, in this case, it doesn't really produce the desired behavior. It would still compile one of the operations (the first one, it appears) as _the_ entry point for the executable. So if the user has run any cells in the notebook with `@EntryPoint()` — or worse, if they had run one earlier and then deleted it — it puts them in a weird state where all of the non-`@EntryPoint()` operations become invisible and can no longer be executed. Only the first one that they happened to compile would be executable.

And I believe ignoring any `@EntryPoint()` attributes within the notebook cells is actually the functionally correct thing to do here, because we'll eventually be auto-generating a new `@EntryPoint` for job execution, and it shouldn't conflict with anything else that happens to be defined in the notebook.

So I guess, to state the core question: is there a way for IQ# to tell the compiler to simply ignore the existence of these attributes? I stepped all the way down to where we call the C# codegen classes, but I didn't see a way to do this (please let me know if there's something I'm missing!). Otherwise I think this might be the best we can do for now.

(And yes, as written, the regex would strip the string `"@EntryPoint()"` out of string literals or comments or whatever. I can't imagine that's common, so it may be ok to do this, but it's obviously not the ideal way.)

Reply:
I was thinking this would be compiled as library, and the error to ignore would be EntryPointInLibrary. Would that make sense?

Reply:
Ok, yes. I was stuck in the mindset of compiling as an executable because (a) we want to compile the workspace (.qs files) as an executable, and (b) we will also need to compile the generated entry point into an executable for job submission. But there is another option: we could the snippets into a library which is then *referenced* by that generated entry point.

I'll try to rework these changes as follows:
- Compile workspace as executable and snippets as library.
- Ignore the error EntryPointInLibrary when compiling snippets.

And when we add the generated entry point, we'll compile it into a third assembly that references both the workspace and the snippets. And this eliminates the need for something hacky like the regex. Does that sound good? Also looping in @anpaz-msft for another opinion (sorry that we're designing this in PR comments).